system/solaris_zone.py

```diff
@@ -0,0 +1,362 @@
+        Solaris 11 kernel zone'
+    required: false
+    default: null
+  timeout:
+    description:
+      - Timeout, in seconds, for zone to boot.
+    required: false
+    default: 600
+'''
+
+EXAMPLES = '''
+# Create a zone, but don't boot it
+solaris_zone: name=zone1 state=present path=/zones/zone1 sparse=true root_password="Be9oX7OSwWoU."
+      config='set autoboot=true, add net, set physical=bge0, set address=10.1.1.1, end'
+
+# Create a zone and boot it
+solaris_zone: name=zone1 state=running path=/zones/zone1 root_password="Be9oX7OSwWoU."
+      config='set autoboot=true, add net, set physical=bge0, set address=10.1.1.1, end'
+
+# Boot an already created zone
+solaris_zone: name=zone1 state=running
+
+# Stop a zone
+solaris_zone: name=zone1 state=stopped
+
+# Destroy a zone
+solaris_zone: name=zone1 state=absent
+'''
+
+class Zone(object):
+    def __init__(self, module):
+        self.changed        = False
+        self.msg            = []
+
+        self.module         = module
+        self.path           = self.module.params['path']
+        self.name           = self.module.params['name']
+        self.sparse         = self.module.params['sparse']
+        self.root_password  = self.module.params['root_password']
+        self.timeout        = self.module.params['timeout']
+        self.config         = self.module.params['config']
+        self.create_options = self.module.params['create_options']
+
+        self.zoneadm_cmd    = self.module.get_bin_path('zoneadm', True)
+        self.zonecfg_cmd    = self.module.get_bin_path('zonecfg', True)
+        self.ssh_keygen_cmd = self.module.get_bin_path('ssh-keygen', True)
+
+    def create(self):
+        if not self.path:
+            self.module.fail_json(msg='Missing required argument: path')
+
+        t = tempfile.NamedTemporaryFile(delete = False)
+
+        if self.sparse:
+            t.write('create %s\n' % self.create_options)
+            self.msg.append('creating sparse root zone')
+        else:
+            t.write('create -b %s\n' % self.create_options)
+            self.msg.append('creating whole root zone')
+
+        t.write('set zonepath=%s\n' % self.path)
+
+        if self.config:
+            for line in self.config:
+                t.write('%s\n' % line)
+        t.close()
+
+        cmd = '%s -z %s -f %s' % (self.zonecfg_cmd, self.name, t.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to create zone. %s' % (out + err))
+        os.unlink(t.name)
+
+        cmd = '%s -z %s install' % (self.zoneadm_cmd, self.name)
+        (rc, out, err) = self.module.run_command(cmd)
+        if rc != 0:
+            self.module.fail_json(msg='Failed to install zone. %s' % (out + err))
+
+        self.configure_sysid()
+        self.configure_password()
+        self.configure_ssh_keys()
+
+    def configure_sysid(self):
```

Comment:
Configuring the zone doesn't work like this with Solaris 11 zones, causing the zone to go into manual configuration and the boot timeout to trip.

On Solaris 11, one instead must pass `-c -c config_profile.xml` to `zoneadm install`.  The XML profile can be created with the `sysconfig create-profile -o ...` command.

For now, would it be best to document that this module doesn't work with Solaris 11 zones?